Assets/Scripts/Core/Utils/WeaponSwitcher.cs

```diff
@@ -0,0 +1,28 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace OperationBlackwell.Core {
+	public class WeaponSwitcher : MonoBehaviour {
+
+		[Header("Sword")]
+		[SerializeField] private GameObject swordButton_;
+		[SerializeField] private Sprite swordButtonTextureSelected_;
+		[SerializeField] private Sprite swordButtonTextureUnselected_;
+
+		[Header("Gun")]
+		[SerializeField] private GameObject gunButton_;
+		[SerializeField] private Sprite gunButtonTextureSelected_;
+		[SerializeField] private Sprite gunButtonTextureUnselected_;
+		public void ChangeActiveWeaponIndicator(int index) {
+			if(index == 0) {
+				swordButton_.GetComponent<Image>().sprite = swordButtonTextureSelected_;
```

Comment:
It's not good for the engine to do this, recommend getting the components once on start or awake and storing that in private variables

Reply:
Noted, I'll PR a fix soon.